pkg/security/api/v1/types.go

```diff
@@ -80,16 +80,17 @@ type PodSecurityPolicyReview struct {
 // PodSecurityPolicyReviewSpec defines specification for PodSecurityPolicyReview
 type PodSecurityPolicyReviewSpec struct {
 	// template is the PodTemplateSpec to check. The template.spec.serviceAccountName field is used
-	// if serviceAccountNames is empty, unless the template.spec.serviceAccountName is empty,
+	// if serviceAccountNames is not nil and empty, unless the template.spec.serviceAccountName is empty,
 	// in which case "default" is used.
-	// If serviceAccountNames is specified, template.spec.serviceAccountName is ignored.
+	// If serviceAccountNames is specified or nil, template.spec.serviceAccountName is ignored.
```

Comment:
@liggitt 
I agree but is it so different from: ` "foo", ["a"]  =  ["a"]` ?
the idea is that `nil` means `all`, regardless of `template.spec.serviceAccountName`.
Finally:
```go
"foo", []    = ["foo"]
"foo", nil   = ["default","builder","deployer",...]
"",    []    = ["default"]
"",    nil   = ["default","builder","deployer",...]
"foo", ["a"] = ["a"]
"",    ["a"] = ["a"]
````
Both choices look quite arbitrary to me, but I'm OK to close it (and fix the bug in another PR).